src/Microsoft.SqlTools.ServiceLayer/Connection/ISqlConnectionFactory.cs

```diff
@@ -16,5 +18,15 @@ public interface ISqlConnectionFactory
         /// Create a new SQL Connection object
         /// </summary>
         DbConnection CreateSqlConnection(string connectionString, string azureAccountToken);
+
+        /// <summary>
+        /// Safely casts DbConnection into SqlConnection
+        /// </summary>
+        SqlConnection ConvertDBConnectionToSqlConnection(DbConnection connection);
+
+        /// <summary>
+        /// Check if authorization is for a contained database user.
+        /// </summary>
+        bool CheckIfContainedAuthorization(DbConnection connection);
```

Comment:
I don't think this is the correct interface for these methods.  Possibly the Convert method could be in a factory interface, but certainly the check method belongs.  Both the convert and check operations aren't creating anything, so problem neither belongs in a factory and should be in a helper\utils class.